control-plane/pkg/reconciler/broker/broker_lifecycle.go

```diff
@@ -140,20 +140,25 @@ func (manager *statusConditionManager) reconciled() reconciler.Event {
 	return reconciledNormal(broker.Namespace, broker.Name)
 }
 
-func (manager *statusConditionManager) failedToUpdateDispatcherPodsAnnotation(broker *eventing.Broker, err error) {
+func (manager *statusConditionManager) failedToUpdateDispatcherPodsAnnotation(err error) {
 
 	// We don't set status conditions for dispatcher pods updates.
 
 	// Record the event.
 	manager.recorder.Eventf(
-		broker,
+		manager.Broker,
 		corev1.EventTypeWarning,
 		"failed to update dispatcher pods annotation",
 		"%v",
 		err,
 	)
 }
 
+func (manager *statusConditionManager) failedToUpdateReceiverPodsAnnotation(err error) reconciler.Event {
+
+	return fmt.Errorf("failed to update receiver pods annotation: %w", err)
```

Comment:
Why you don't record the event here?

Reply:
The generated reconciler records errors returned by `ReconcileKind` or `FinalizeKind` with reason "InternalError" and the message is the error message.

Reply:
Ok